packages/flutter_tools/lib/src/macos/cocoapods.dart

```diff
@@ -375,25 +380,41 @@ class CocoaPods {
     }
   }
 
-  // Previously, the Podfile created a symlink to the cached artifacts engine framework
-  // and installed the Flutter pod from that path. This could get out of sync with the copy
-  // of the Flutter engine that was copied to ios/Flutter by the xcode_backend script.
-  // It was possible for the symlink to point to a Debug version of the engine when the
-  // Xcode build configuration was Release, which caused App Store submission rejections.
-  //
-  // Warn the user if they are still symlinking to the framework.
   void _warnIfPodfileOutOfDate(XcodeBasedProject xcodeProject) {
     if (xcodeProject is! IosProject) {
       return;
     }
+
+    // Previously, the Podfile created a symlink to the cached artifacts engine framework
```

Comment:
Is there potentially an end-date for these mitigations, or is this code needed forever?

Reply:
The primary purpose of https://github.com/flutter/flutter/pull/59044 is to move most of this logic into the tool so we can change it without requiring the user to regenerate their Podfile.  It imports a new Ruby helper (bleh) from flutter_tools/bin and then calls those helper methods.  So, 🤞 this is one of the last migration prompts we ever need (knock on wood as well).

So this particular migration prompt is to get them to generate the new version of the Podfile that imports from the tool.  It will be needed long as we want to support projects created in 1.17 that upgrade to latest.